griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,293 @@
+        )
+        delegated_credentials = credentials.with_subject(self.owner_email)
+        service = build('drive', 'v3', credentials=delegated_credentials)
+
+        return service
+
+    def _path_to_file_id(self, service, path: str) -> Optional[str]:
+        parts = path.split('/')
+        current_id = 'root'
+
+        for index, part in enumerate(parts):
+            if index == len(parts) - 1:
+                query = f"name='{part}' and '{current_id}' in parents"
+            else:
+                query = f"name='{part}' and '{current_id}' in parents and mimeType='application/vnd.google-apps.folder'"
+
+            response = service.files().list(q=query).execute()
+            files = response.get('files', [])
+
+            if not files:
+                return None
+            current_id = files[0]['id']
+
+        return current_id
+
+    @activity(config={
+        "description": "List files in a specific Google Drive folder or the root if no folder is specified",
+        "schema": Schema({
+            Optional("folder_path",
+                     description="Path of the Google Drive folder (like 'MainFolder/Subfolder1/Subfolder2') "
+                                 "from which files should be listed. If not provided, files from the root "
+                                 "directory will be listed."): str,
+            Optional("max_files",
+                     description="Maximum number of files to list. If not provided, the default page size will be used."): int
+        })
+    })
+    def list_files(self, params: dict) -> ListArtifact | ErrorArtifact:
+        values = params["values"]
+
+        try:
+            service = self._build_client(self.LIST_FILES_SCOPES)
+
+            folder_id = None
+            if "folder_path" in values and values["folder_path"]:
+                folder_id = self._path_to_file_id(service, values["folder_path"])
+                if not folder_id:
+                    return ErrorArtifact(f"Could not find folder: {values['folder_path']}")
+
+            if folder_id:
+                query = f"'{folder_id}' in parents and trashed=false"
+            else:
+                query = "mimeType != 'application/vnd.google-apps.folder' and 'root' in parents and trashed=false"
+
+            user_max_files = values.get("max_files")
+            max_files = self.page_size if user_max_files is None else user_max_files
+
+            items = []
+            next_page_token = None
+
+            while True:
+                results = service.files().list(
+                    q=query,
+                    pageSize=max_files,
+                    pageToken=next_page_token,
+                ).execute()
+
+                files = results.get('files', [])
+                items.extend(files)
+
+                next_page_token = results.get('nextPageToken')
+
+                if not next_page_token:
+                    break
+
+            return ListArtifact(items)
```

Comment:
`items` here should be a list of `TextArtifact`s. Take a loot at the `ListArtifact.value` signature.